src/mono/wasi/build/WasiApp.Native.targets

```diff
@@ -359,74 +359,43 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="_GenerateAssemblyObjectFiles" Returns="@(_WasiObjectFilesForBundle)"
-    Condition="'$(WasmSingleFileBundle)' == 'true'"
-  >
-    <!-- Get the file hashes of everything in @(_WasmBundleFiles), then pass it all to EmitWasmBundleObjectFiles. This
-         will emit corresponding .o files for anything we don't already have on disk. -->
-    <PropertyGroup>
-      <_WasmAssembliesBundleObjectFile>wasi_bundled_assemblies.o</_WasmAssembliesBundleObjectFile>
-      <_WasmIcuBundleObjectFile>wasi_bundled_icu.o</_WasmIcuBundleObjectFile>
-      <_WasmRuntimeConfigBundleFile>wasi_runtime_config_bin.o</_WasmRuntimeConfigBundleFile>
-    </PropertyGroup>
-    <!-- TODO make this incremental compilation -->
-    <EmitBundleObjectFiles
-      FilesToBundle="@(_WasmAssembliesInternal)"
-      ClangExecutable="$(WasiClang)"
-      BundleRegistrationFunctionName="mono_register_assemblies_bundle"
-      BundleFile="$(_WasmAssembliesBundleObjectFile)"
-      OutputDirectory="$(_WasmIntermediateOutputPath)">
-      <Output TaskParameter="BundledResources" ItemName="_WasmBundledAssemblies" />
-    </EmitBundleObjectFiles>
-
-    <ItemGroup>
-      <_WasiObjectFilesForBundle Include="$(_WasmIntermediateOutputPath)$(_WasmAssembliesBundleObjectFile)" />
-      <_WasiObjectFilesForBundle Include="%(_WasmBundledAssemblies.DestinationFile)" />
-    </ItemGroup>
-
-    <!-- Clean up the bundle-objects dir - remove anything we no longer need -->
+  <Target Name="_GenerateObjectFilesForSingleFileBundle" Returns="@(_WasiObjectFilesForBundle)" Condition="'$(WasmSingleFileBundle)' == 'true'">
     <ItemGroup>
-      <WasmBundleFileToDelete Include="$(_WasmIntermediateOutputPath)bundled_*.o" />
-      <WasmBundleFileToDelete Remove="$(_WasmIntermediateOutputPath)$(_WasmAssembliesBundleObjectFile)" />
-      <WasmBundleFileToDelete Remove="%(_WasmBundledAssemblies.DestinationFile)" />
-    </ItemGroup>
-
-    <EmitBundleObjectFiles
-      Condition="'$(InvariantGlobalization)' != 'true'"
-      FilesToBundle="$(MicrosoftNetCoreAppRuntimePackRidNativeDir)icudt.dat"
-      ClangExecutable="$(WasiClang)"
-      BundleRegistrationFunctionName="mono_register_icu_bundle"
-      BundleFile="$(_WasmIcuBundleObjectFile)"
-      OutputDirectory="$(_WasmIntermediateOutputPath)">
-      <Output TaskParameter="BundledResources" ItemName="BundledWasmIcu" />
-    </EmitBundleObjectFiles>
-
-    <ItemGroup Condition="'$(InvariantGlobalization)' != 'true'">
-      <_WasiObjectFilesForBundle Include="$(_WasmIntermediateOutputPath)$(_WasmIcuBundleObjectFile)" />
-      <_WasiObjectFilesForBundle Include="%(BundledWasmIcu.DestinationFile)" />
-
-      <WasmBundleFileToDelete Remove="$(_WasmIntermediateOutputPath)$(_WasmIcuBundleObjectFile)" />
-      <WasmBundleFileToDelete Remove="%(BundledWasmIcu.DestinationFile)" />
+      <_EmitBundleItemName
+                Include="_WasmAssembliesInternal"
```

Comment:
Do we no longer need to reference ItemGroups by `@(ItemGroupName)`?

Reply:
The `include` here only references to the item *name*, which we then use to get the real item like ` FilesToBundle="@(%(_EmitBundleItemName.Identity))"`. But I was adding this when I was still trying to do a different style for the task itself. I'll check if this can be dropped now.

Reply:
Simplified it.